lib/basin-util.c

```diff
@@ -95,6 +112,26 @@ basin_override_resources (const gchar *content)
     }
   xmlXPathFreeObject(objects);
 
+  /* override videos links */
+  objects = xmlXPathEvalExpression ("//a[@data-libingester-asset-id]", context);
+  if (!xmlXPathNodeSetIsEmpty (objects->nodesetval))
+    {
+      htmlDocPtr img_node = htmlReadDoc (THUMBNAIL_SVG, "", NULL, HTML_PARSE_RECOVER | HTML_PARSE_NOERROR);
+      xmlNodeSetPtr nodeset = objects->nodesetval;
+      for (int i = 0; i < nodeset->nodeNr; i++)
+        {
+          ekn_id = xmlGetProp (nodeset->nodeTab[i], "data-libingester-asset-id");
+          prop = g_strdup_printf ("ekn:///%s", ekn_id);
+          xmlSetProp (nodeset->nodeTab[i], "href", prop);
+          xmlFree (prop);
```

Comment:
Should be `g_free()` — probably both call `free()` in practice, but always free API-allocated memory with the free function advised by that API

Reply:
(And this is a pre-existing defect, but strictly speaking the elements of the list returned by `basin_find_resources()` should be freed with `xmlFree()`, but GJS has no way to know that ... so you should probably copy them with `g_strdup()` and free the original XML characters)